### DISCUSSION TITLE
Is it possible to target a single SketchyBar across multiple monitors?

### DISCUSSION THREAD

**Discussion by rogeruiz** [FelixKratz/SketchyBar#326] [Q&A] [answered]
Hi @FelixKratz,

I found #74 and was wondering if there's a way to programmatically hide and show certain items in the configuration based on the width of the current screen that the bar is displayed on? I have a few vertifical monitors with very different resolutions. And I find that my widths sometimes prevent things from showing up. So I'd like to dynamically update the contents of my items in SketchyBar depending on the screen size or even the monitor that it's displaying on. 

Is this something that's possible today?

For now, I've just put all items I would like to prioritize on the right side so that when things do get squished the backgrounds of the right-side items draw over the left-side items. This is good enough for now, but I would for a way to hide and show certain items with the same functionality in #74 but based on some kind of custom event.

I'd expect that a lot of this functionality would need to be written in a plugin, so I'm mostly looking for some validation of the hypothesis that I can achieve this today with a plugin. My main concern with just writing a plugin is that the way SketchyBar is rendered today is that the configuration is shared across all the monitors I have.

So is there a way to target a single render of SketchyBar across monitors?

**Comment by FelixKratz** [comment, marked as answer]
This is something that is possible today, I think what you are looking for is the `associated_display` option that every item has, it can be used to show certain items only on some specific display, e.g. if I want an item to show only on display 1 I would set:
```bash
sketchybar --set <name> associated_display=1
```
and if I wanted to show the item on display 1 and 3 instead I would do it this way:
```bash
sketchybar --set <name> associated_display=1,3
```
The same is possible for mission control spaces via the option `associated_space`.
This way you can have completely different items showing on different displays or even spaces.

**Reply by rogeruiz** [reply]
Nice thanks! I'll try this out in my setup and hopefully be able to answer my original question with a plug-in/solution example setup.

**Reply by rogeruiz** [reply]
Alright, as I tried to customize things using event driven and some hacky `systemprofiler` calls, I realized that it's okay to hard-code the `associated_display` on elements that you would like to only appear on certain displays. 

For other folks reading this, you're mileage may vary and you may want to have a more elegant solution. But for me, it worked out that I only ever want to disable a single monitor's output. The above solution works perfectly.
